### PR TITLE
[MOB-469] PoA Analytics

### PR DESCRIPTION
**Pull request by franciscoCalado**
**What does this PR do?**

This PR added Rakam analytics for PoA;

**Database changed?**

No

**Where should the reviewer start?**

- [ ] PoaAnalytics.java

**How should this be manually tested?**

Make sure the events in the ticket are being sent;

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-469](https://aptoide.atlassian.net/browse/MOB-469)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java [hunk 1]
```diff
@@ -501,6 +508,12 @@ public void handlePoaCompletedToSendEvent() {
                 .equals(ProofStatus.COMPLETED)) {
               analytics.sendPoaCompletedEvent(proof.getPackageName(), proof.getCampaignId(),
                   Integer.toString(proof.getChainId()));
+            } else if (!proof.getProofStatus()
+                .equals(ProofStatus.PROCESSING) || !proof.getProofStatus()
+                .equals(ProofStatus.SUBMITTING) || !proof.getProofStatus()
+                .equals(ProofStatus.COMPLETED)) {
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java]: Can't this just be an else?
Processing and submitting are not "isTerminate" so if the chain reaches here we know that it wasn't with those 2 cases. And this else is from if(COMPLETED) so no need for that also

**Review by JDSM01 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java [hunk 2]
```diff
@@ -493,6 +493,13 @@ public void handlePoaCompletedToSendEvent() {
     if (completedEventDisposable == null || completedEventDisposable.isDisposed()) {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java:494]: Can't we handle both scenarios in the same place?
Also, can we extract the logic to a private method so the main method continues readable?

**Review by JDSM01 — CHANGES_REQUESTED** [review]
Conflicts
Shouldn't we also add an error event on line 121 of the WalletPoaService.java?

File: app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java (hunk 2 above)
Comment (carvalho2707) [reply app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java:494]: this applies for the whole method

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

**Review by JDSM01 — APPROVED** [review]

**Pull request merged**